compressor/filters/base.py

```diff
@@ -208,5 +211,26 @@ def input(self, **kwargs):
                 self.infile.close()
             if self.outfile is not None:
                 self.outfile.close()
-
         return smart_text(filtered)
+
+
+class CachedCompilerFilter(CompilerFilter):
+
+    def __init__(self, mimetype, **kwargs):
+        self.mimetype = mimetype
+        super(CachedCompilerFilter, self).__init__(**kwargs)
+
+    def input(self, **kwargs):
+        if self.mimetype in settings.COMPRESS_CACHEABLE_PRECOMPILERS:
+            key = self.get_cache_key()
+            data = cache.get(key)
+            if data:
```

Comment:
Might want to check for `not None` instead to avoid recompiling empty files here